scraper.rb

```diff
@@ -3,43 +3,29 @@
 # frozen_string_literal: true
 
 require 'pry'
+require 'require_all'
 require 'scraped'
 require 'scraperwiki'
 
+require_rel 'lib'
+
 # require 'open-uri/cached'
 # OpenURI::Cache.cache_path = '.cache'
 require 'scraped_page_archive/open-uri'
 
-def noko_for(url)
-  Nokogiri::HTML(open(url).read)
+def scrape(h)
+  url, klass = h.to_a.first
+  klass.new(response: Scraped::Request.new(url: url).response)
 end
 
-def scrape_list(url)
-  noko = noko_for(url)
-  noko.xpath('.//ul[@id="leftnavigation"]/li[contains(.,"States Members")]/following-sibling::li[@class="child"]//a/@href').each do |li|
-    scrape_mp(li.text)
+def scrape_list(term, url)
+  (scrape url => MembersPage).member_urls.each do |mem_url|
+    data = (scrape mem_url => MemberPage).to_h.merge(district: 'Alderney',
+                                                     party:    'Independent',
+                                                     term:     term)
+    ScraperWiki.save_sqlite(%i(id term), data)
   end
 end
 
-def scrape_mp(url)
-  noko = noko_for(url)
-  data = {
-    id:       url[%r{/article/(\d+)/}, 1],
-    name:     noko.css('div#contentwrap h1').text.tidy,
-    image:    noko.css('div.limage img/@src').text,
-    district: 'Alderney',
-    party:    'Independent',
-    term:     2014,
-    source:   url,
-  }
-  # if matched = noko.css('div#contentwrap').text.match(/Elected to Office in (.*?) (20\d\d)/)
-  # data[:start_date] = Date.parse("%s-%s-01" % matched.captures.reverse).to_s
-  # end
-  data[:image] = URI.join(@BASE, URI.escape(data[:image])).to_s unless data[:image].to_s.empty?
-  ScraperWiki.save_sqlite(%i(id term), data)
-end
-
-@BASE = 'http://www.alderney.gov.gg'
-
 ScraperWiki.sqliteexecute('DELETE FROM data') rescue nil
-scrape_list('http://www.alderney.gov.gg/article/4077/States-Members')
+scrape_list(2014, 'http://www.alderney.gov.gg/article/4077/States-Members')
```

Comment:
There doesn't seem to be much value in having a separate `scrape_list` here any more, especially if you're changing its signature anyway